src/containers/AssetForm.js

```diff
@@ -1,16 +1,416 @@
+            <div className="App-grid-item">
+              <TextField
+                hintText="Other retention period"
+                disabled={this.state.retention !== 'other'}
+                value={this.state.retention_other === null ? "" : this.state.retention_other}
+                onChange={(e, value) => this.handleChange('retention_other', value)}
+              />
+            </div>
+          </div>
+
+          <CheckboxGroup
+            title='What are the risks of holding this information asset?'
+            labels={RISK_TYPE_LABELS}
+            values={this.state.risk_type}
+            onChange={(value) => this.handleChange('risk_type', value)}
+          />
+
+          <div className="App-grid-container App-grid-1">
+            <div className="App-grid-item">
+              <TextField
+                hintText="Where is the asset stored?"
+                value={this.state.storage_location === null ? "" : this.state.storage_location}
+                onChange={(e, value) => this.handleChange('storage_location', value)}
+              />
+            </div>
+          </div>
+
+          <div className="App-grid-container App-grid-2">
+            <div className="App-grid-item">
+              What format is the asset stored in?
+            </div>
+            <div className="App-grid-item">
+              <RadioButtonGroup
+                name="storage_format"
+                valueSelected={this.state.storage_format.sort().toString()}
+                onChange={(event, value) => this.setState({storage_format: value.split(',')})}
+                style={storageFormatGroupStyle}
+              >
+                <RadioButton value="digital" label="Digital" style={storageFormatButtonStyle} />
+                <RadioButton value="paper" label="Paper" style={storageFormatButtonStyle} />
+                <RadioButton value="digital,paper" label="Both" style={storageFormatButtonStyle} />
+              </RadioButtonGroup>
+            </div>
+          </div>
+
+          <CheckboxGroup
+            title='What are the risks of holding this information asset?'
+            labels={DIGITAL_STORAGE_SECURITY_LABELS}
+            values={this.state.digital_storage_security}
+            onChange={(value) => this.handleChange('digital_storage_security', value)}
+            disabled={this.state.storage_format.indexOf("digital") === -1}
+          />
+          <CheckboxGroup
+            labels={PAPER_STORAGE_SECURITY_LABELS}
+            values={this.state.paper_storage_security}
+            onChange={(value) => this.handleChange('paper_storage_security', value)}
+            disabled={this.state.storage_format.indexOf("paper") === -1}
+          />
+        </div>
+      </div>
+    )
+  }
+}
+
+const RoutedAssetForm = (props) => (
+  <Route path="/asset/:asset"
+         render={(routeProps) => <AssetForm handleMessage={props.handleMessage.bind(this)} {...routeProps}/>} />
```

Comment:
This file should be renamed if it actually exposes a Route. What's the rationale for this BTW? It would seem to me that one should separate Route-s from the components which they render?